### PR TITLE
Marcos.cifuentes/feature card list activity ui

### PR DESCRIPTION
**Pull request by MarcosCGlobant**
In this PR I implemented the UI for the card list activity feature, the information shown is mocked, because the next PR will be the service implementation.

- Modified fragments to add onClickListeners to the elements.
- Added Card List implementation with mocked data
- Unit test for the view model

GIF of the app:

card lists ui

Unit test passing and coverage:

test passed cardList

cardlist coverage

### CODE REVIEW

File: app/src/main/java/com/globant/pokemontcgapp/fragment/PokemonSupertypeFragment.kt [hunk 1]
```diff
@@ -51,17 +55,30 @@ class PokemonSupertypeFragment : Fragment() {
...
-                    context,
-                    resources.configuration.getColumnsByOrientation(
-                        COLUMNS_PORTRAIT,
-                        COLUMNS_LANDSCAPE
-                    )
+        pokemonSupertypes.let {
+            val pokemonSupertypesAdapter = PokemonSecondaryTypesAdapter(pokemonSupertypes) { pokemonSupertypeElement ->
+                startActivity(
+                    context?.let { it1 ->
+                        PokemonCardListActivity.getIntent(it1).apply {
```
Comment (SartiMau) [inline app/src/main/java/com/globant/pokemontcgapp/fragment/PokemonSupertypeFragment.kt]: create a method that return the intent with all the extras in the PokemonCardListActivity

File: app/src/main/java/com/globant/pokemontcgapp/fragment/PokemonTypeFragment.kt [hunk 2]
```diff
@@ -51,11 +55,24 @@ class PokemonTypeFragment : Fragment() {
...
-        binding.pokemonAlltypesRecyclerView.apply {
-            layoutManager =
-                GridLayoutManager(context, resources.configuration.getColumnsByOrientation(COLUMNS_PORTRAIT, COLUMNS_LANDSCAPE))
-            adapter = PokemonTypesAdapter(pokemonTypes)
-            visibility = View.VISIBLE
+        pokemonTypes.let {
+            val pokemonTypesAdapter = PokemonTypesAdapter(pokemonTypes) { pokemonTypeElement ->
+                startActivity(
+                    context?.let { it1 ->
+                        PokemonCardListActivity.getIntent(it1).apply {
```
Comment (SartiMau) [inline app/src/main/java/com/globant/pokemontcgapp/fragment/PokemonTypeFragment.kt]: same

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/globant/pokemontcgapp/fragment/PokemonSupertypeFragment.kt (hunk 1 above)
Comment (MarcosCGlobant) [reply app/src/main/java/com/globant/pokemontcgapp/fragment/PokemonSupertypeFragment.kt]: done, thanks

File: app/src/main/java/com/globant/pokemontcgapp/fragment/PokemonTypeFragment.kt (hunk 2 above)
Comment (MarcosCGlobant) [reply app/src/main/java/com/globant/pokemontcgapp/fragment/PokemonTypeFragment.kt]: done, thanks

**Review by SartiMau — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Review by hernanhorquin96 — APPROVED** [review]

File: app/src/main/java/com/globant/pokemontcgapp/adapter/PokemonTypesAdapter.kt [hunk 3]
```diff
@@ -28,11 +30,12 @@ class PokemonTypesAdapter(private val pokemonTypes: List<PokemonType>) : Recycle
 
     override fun getItemCount(): Int = pokemonTypes.size
 
-    class ViewHolder(itemView: View) :
+    class ViewHolder(itemView: View, private val onPokemonTypeClicked: (PokemonType) -> Unit) :
```
Comment (mauroLonghi) [inline app/src/main/java/com/globant/pokemontcgapp/adapter/PokemonTypesAdapter.kt]: Marcos, please change the usage of this callback...Try to define a ListenerIterface to represent the click callback.

Comment (MarcosCGlobant) [reply app/src/main/java/com/globant/pokemontcgapp/adapter/PokemonTypesAdapter.kt]: refactored to use listeners interface, thanks

**Review by SartiMau — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Review by hernanhorquin96 — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Pull request merged**